### PR TITLE
Laravel 10.x Shift

### PR DESCRIPTION
**Pull request by Jellyfrog**
This pull request includes the changes for upgrading to Laravel 10.x. Feel free to commit any additional changes to the `shift-89094` branch.

**Before merging**, you need to:

- Checkout the `shift-89094` branch
- Review **all** pull request comments for additional changes
- Run `composer update` (if the scripts fail, add `--no-scripts`)
- Clear any config, route, or view cache
- Thoroughly test your application ([no tests?](https://laravelshift.com/laravel-test-generator), [no CI?](https://laravelshift.com/ci-generator))

If you need help with your upgrade, check out the [Human Shifts](https://laravelshift.com/human-shifts). You may also join the [Shifty Coders](https://laravelshift.com/shifty-coders) Slack workspace to level-up your Laravel skills.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
:information_source: Shift noticed you have additional namespaces in your application. You may use the [Consolidate Namespaces Shift](https://laravelshift.com/laravel-consolidate-custom-namespaces) to simplify your namespaces into the default Laravel `App` namespace.

**Comment by Jellyfrog** [comment]
:x: Laravel 9 removed its dependency for `fruitcake/laravel-cors`. This behavior is now provided by the framework through `Illuminate\Http\Middleware\HandleCors`.

Shift found references to `Fruitcake\Cors` in the following files. You should review these references and adopt the internal middleware.

- [ ] app/Http/Middleware/HandleCors.php

**Comment by Jellyfrog** [comment]
:x: Shift could not upgrade the following files since they differed from the default Laravel version. You will need to compare these files against the [default Laravel 10 versions](https://github.com/laravel-shift/laravel-10.x/) and merge any changes:

- [ ] lang/en/validation.php

**Comment by Jellyfrog** [comment]
:warning: Shift upgraded your configuration files by defaulting them and merging your _true customizations_. These include values which are not changeable through core `ENV` variables.

You should [review this commit](https://github.com/librenms/librenms/pull/14995/commits/59598876e6abab60b169de590c6790dfb00ed599?w=1) for additional customizations or opportunities to use new `ENV` variables. If you have a lot of customizations, you may undo this commit with `git revert 59598876` and make the [config file changes](https://github.com/laravel-shift/laravel-9.x/pull/4/commits/5401110f1f41793ecdce268495768abfa4a53dfe) manually.

**Comment by Jellyfrog** [comment]
:information_source: Shift [updated your dependencies](https://github.com/librenms/librenms/pull/14995/commits/ee3bda12aef0143cea9b5cb099ab17151e75e9e2?w=1) for Laravel 10. While many of the popular packages are reviewed, you may have to update additional packages in order for your application to be compatible with Laravel 10. Watch [dealing with dependencies](https://laravelshift.com/videos/update-incompatible-composer-dependencies) for tips on handling any Composer issues.

The following dependencies were updated by a major version and may have their own changes. You may check their changelog for any additional upgrade steps.

- [ ] [doctrine/dbal](https://packagist.org/packages/doctrine/dbal)
- [ ] [librenms/laravel-vue-i18n-generator](https://packagist.org/packages/librenms/laravel-vue-i18n-generator)
- [ ] [php-flasher/flasher-laravel](https://packagist.org/packages/php-flasher/flasher-laravel)

**Comment by Jellyfrog** [comment]
:information_source: Shift detected your application has a test suite. To make verifying the upgrade to Laravel 10 easier, Shift did not upgrade your test suite to PHPUnit 10. Once you have completed your upgrade, you may run the [PHPUnit 10 Shift](https://laravelshift.com/upgrade-phpunit-10) for free to upgrade your test suite to PHPUnit 10 separately.

**Comment by Jellyfrog** [comment]
:information_source: Laravel 9 adopted [anonymous migrations](https://laravel-news.com/laravel-anonymous-migrations). Shift [automated this change](https://github.com/librenms/librenms/pull/14995/commits/dc491d1fd8469cb6beb73ba99ead3a3c805b1493?w=1) to align with modern Laravel conventions and avoid naming migrations.

**Comment by Jellyfrog** [comment]
:warning: Laravel 10 has added a return type of `array` to the Event `broadcastOn` and Notification `via` methods. However, you may still return a single channel from these methods.

Shift added the `array` return type to these methods in the following classes. You should review them to ensure you are returning an `array`, or remove the return type.

- [ ] app/Events/DevicePolled.php
- [ ] app/Events/PollingDevice.php
- [ ] app/Notifications/AlertNotification.php

**Comment by Jellyfrog** [comment]
:information_source: Laravel 10 added PHP type hints to all user-land code included in a new Laravel application. In an effort to modernize your code, Shift [added type hints](https://github.com/librenms/librenms/pull/14995/commits/83737d3fdaebe014c530e7c3d17e8fd9a4053162?w=1) to any method which is used by Laravel.

**Comment by Jellyfrog** [comment]
:information_source: Now with type hints in your code, defining types within PHP DocBlocks is redundant. Laravel has removed all of the `@param` and `@return` tags from its DocBlocks where types are defined with PHP. Similarly, Shift [removed these tags](https://github.com/librenms/librenms/pull/14995/commits/f3762ca567c72704fc47300edc9812f56bd53c51?w=1) from any DocBlock where the code now has _equivalent_ type hints.

**Comment by Jellyfrog** [comment]
:information_source: Shift understands developers have different preferences when it comes to type hints. All of Shift's automation is done in nice, atomic commits. This makes it easier to undo any of the changes Shift makes.

If you wish to undo the changes relating to type hints, you may run:

- `git revert f3762ca56` to revert the DocBlock changes.
- `git revert 9d99e3e7d` to revert the type hints added from DocBlocks.
- `git revert 83737d3fd` to revert the type hints added for Laravel 10.

**Comment by Jellyfrog** [comment]
:warning: In Laravel 10 you may no longer directly cast a `DB:raw()` expression to a string. Instead, you must retrieve the value from the expression using the `getValue()` method.

Shift detected calls to `DB::raw()`. You should review these instances to see if you are casting the result to a string.

- [ ] LibreNMS/Alerting/QueryBuilderFluentParser.php
- [ ] LibreNMS/DB/Schema.php
- [ ] LibreNMS/Poller.php
- [ ] LibreNMS/Util/Stats.php
- [ ] app/Http/Controllers/Device/Tabs/LatencyController.php
- [ ] app/Http/Controllers/Table/AlertScheduleController.php
- [ ] app/Http/Controllers/Table/DeviceController.php
- [ ] app/Http/Controllers/Table/LocationController.php
- [ ] app/Http/Controllers/Table/PortsController.php
- [ ] app/Http/Controllers/Widgets/ComponentStatusController.php
- [ ] app/Http/Controllers/Widgets/ServerStatsController.php
- [ ] app/Listeners/MarkNotificationsRead.php
- [ ] app/Models/AlertSchedule.php
- [ ] app/Models/Notification.php
- [ ] app/Models/Poller.php
- [ ] app/Models/PollerCluster.php
- [ ] database/migrations/2018_07_03_091314_create_applications_table.php
- [ ] database/migrations/2018_07_03_091314_create_device_mibs_table.php
- [ ] database/migrations/2018_07_03_091314_create_device_oids_table.php
- [ ] database/migrations/2018_07_03_091314_create_devices_attribs_table.php
- [ ] database/migrations/2018_07_03_091314_create_mibdefs_table.php
- [ ] database/migrations/2018_07_03_091314_create_sensors_table.php
- [ ] database/migrations/2018_07_03_091314_create_state_translations_table.php
- [ ] database/migrations/2019_08_28_105051_fix-template-linefeeds.php
- [ ] database/migrations/2019_12_16_140000_create_customoids_table.php
- [ ] database/migrations/2020_04_08_172357_alert_schedule_utc.php
- [ ] database/migrations/2020_05_30_162638_remove_mib_polling_tables.php
- [ ] database/migrations/2020_09_18_230114_create_service_templates_table.php
- [ ] tests/DBSetupTest.php

**Comment by Jellyfrog** [comment]
:warning: Laravel 10 has a dependency for `monolog/monolog` of 3.x. Shift detected references to `Monolog` classes within the following files. You should review Monolog's [upgrade guide](https://github.com/Seldaek/monolog/blob/main/UPGRADE.md) for any additional changes.

- [ ] app/Logging/CliColorFormatter.php
- [ ] app/Logging/NoColorFormatter.php

**Comment by Jellyfrog** [comment]
:information_source: All of the underlying Symfony components used by Laravel have been upgraded to Symfony 6.2. Shift detected references to Symfony classes within your application. You should review the [Symfony change log](https://github.com/symfony/symfony/blob/6.2/UPGRADE-6.2.md) for any additional changes.

**Comment by Jellyfrog** [comment]
:information_source: Laravel began using Vite to build frontend assets in Laravel 9.19. While you may continue to use Laravel Mix, it is no longer the default. If you wish to modernize your application to use Vite, you may run the [Vite Converter](https://laravelshift.com/convert-laravel-mix-to-vite) for free.

**Comment by Jellyfrog** [comment]
:warning: Laravel 10 requires Composer 2.2 or higher. You should verify the Composer version in your environments by running `composer --version` to ensure it meets this new requirement. If necessary, run `composer self-update` to update Composer.

**Comment by Jellyfrog** [comment]
:tada: Congratulations, you're now running the latest version of Laravel!

Next, you may optionally run the following _Shifts_ to ensure your application is fully upgraded, adopts the latest Laravel conventions, and easier to maintain in the future:

- [Upgrade Checker](https://laravelshift.com/laravel-upgrade-checker) ensures your application is fully upgraded by detecting any outdated code.
- [Laravel Fixer](https://laravelshift.com/laravel-code-fixer) automatically updates your code to the latest Laravel conventions.
- [Tests Generator](https://laravelshift.com/laravel-test-generator) intelligently generates model factories, HTTP Tests, and configuration for your application.
- [CI Generator](https://laravelshift.com/ci-generator) intelligently generates CI jobs to lint PHP, check code style, and run tests, including Dusk.

You may also use the [Shift Workbench](https://laravelshift.com/workbench) to automate common tasks for maintaining your Laravel and PHP applications.

**Comment by Jellyfrog** [comment]
:x: PHP syntax errors were detected after running your Shift. Often these are simply differences between the PHP version on the Shift server (8.1) and your project. Occasionally they are misplaced lines or duplicate import statements.

You may quickly check the PHP syntax locally by running `php -l` on the following files:

- [ ] app/Console/Commands/AddUserCommand.php
- [ ] app/Console/Commands/DevCheckCommand.php
- [ ] app/Console/Commands/GetConfigCommand.php
- [ ] app/Console/Commands/SetConfigCommand.php
- [ ] app/Console/Commands/TranslationGenerateCommand.php
- [ ] app/Providers/RouteServiceProvider.php

**Comment by Jellyfrog** [comment]
- [ ] https://github.com/Seldaek/monolog/blob/main/UPGRADE.md

File: config/database.php [hunk 1]
```diff
@@ -178,32 +179,32 @@
...
+    | Redis is an open source, fast, and advanced key-value store that also
+    | provides a richer body of commands than a typical key-value system
+    | such as APC or Memcached. Laravel makes it easy to dig right in.
+    |
+    */
 
     'redis' => [
 
-        'client' => env('REDIS_CLIENT', 'predis'),
+        'client' => env('REDIS_CLIENT', 'phpredis'),
```
Comment (murrant) [inline config/database.php:207]: probably need to leave this to the predis driver.

File: config/database.php [hunk 2]
```diff
@@ -46,7 +45,7 @@
         'sqlite' => [
             'driver' => 'sqlite',
             'url' => env('DATABASE_URL'),
-            'database' => env('DB_DATABASE', storage_path('librenms.sqlite')),
+            'database' => env('DB_DATABASE', database_path('librenms.sqlite')),
```
[the remark this answers is not in the document]
Comment (Jellyfrog) [reply config/database.php:48]: id say its fine since tests passes :)

**Review by murrant — DISMISSED** [review]
lgtm

**Review by murrant — APPROVED** [review]

**Pull request merged**